.github/workflows/build.yml

```diff
@@ -187,7 +187,10 @@ jobs:
       # This naming convention will be used ONLY for per-commit dev images
       - name: Set docker dev tag
         run: |
-          echo "dev_tag=${{ env.version }}" >> $GITHUB_ENV
+          echo "full_dev_tag=${{ env.version }}"
+          echo "full_dev_tag=${{ env.version }}" >> $GITHUB_ENV
+          echo "minor_dev_tag=$(echo ${{ env.version }}| sed -E 's/([0-9]+\.[0-9]+)\.[0-9]+(-[0-9a-zA-Z\+]+)?$/\1\2/')" 
+          echo "minor_dev_tag=$(echo ${{ env.version }}| sed -E 's/([0-9]+\.[0-9]+)\.[0-9]+(-[0-9a-zA-Z\+]+)?$/\1\2/')" >> $GITHUB_ENV
```

Comment:
Re: async conversation about making this consistent everywhere, I think we need to allow literal `.` as well [for FIPS](https://hub.docker.com/layers/hashicorppreview/consul-enterprise/1.16.0-dev-ent.fips1402-dev-ubi/images/sha256-471fd3d01063226f9b6e400aae8b3ad1262c95ae673b572ecb87e93bf86b52b8?context=explore) in `consul-enterprise`; might be simplest just to allow for that here and in other regexes too.

Reply:
I think this line is where the `.` sneaks in (I remembered bc this bit me in some other situation during 1.16 RC): https://github.com/hashicorp/consul-enterprise/blob/d45b3bbf1085ed52de6015a6c780a9776bb500ab/.github/workflows/build.yml#L533

Reply:
Some of the other repos have the `.`, so I will add.